ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/parser.rs

```diff
@@ -0,0 +1,289 @@
+                // Write image to GCS
+                match write_image_to_gcs(
+                    self.ts,
+                    format,
+                    self.bucket.clone(),
+                    self.entry.token_data_id.clone(),
+                    new_img,
+                )
+                .await
+                {
+                    Ok(filename) => {
+                        self.model.cdn_image_uri =
+                            Some(format!("http://34.160.26.161/{}", filename));
+                        info!(
+                            last_transaction_version = self.entry.last_transaction_version,
+                            "Successfully saved image"
+                        );
+                    },
+                    Err(e) => error!(
+                        last_transaction_version = self.entry.last_transaction_version,
+                        "{}",
+                        e.to_string()
+                    ),
+                }
+            },
+            Err(e) => {
+                self.model.image_optimizer_retry_count += 1;
+                error!(
+                    last_transaction_version = self.entry.last_transaction_version,
+                    "{}",
+                    e.to_string()
+                )
+            },
+        };
+
+        // Optimize animation if exists
+        if let Some(animation_uri) = animation_uri_option {
+            match self.optimize_image(animation_uri).await {
+                Ok((new_img, format, raw_animation_uri)) => {
+                    info!(
+                        last_transaction_version = self.entry.last_transaction_version,
+                        "Successfully optimized image"
+                    );
+
+                    self.model.raw_animation_uri = Some(raw_animation_uri);
+
+                    // Write animation to GCS
+                    match write_image_to_gcs(
+                        self.ts,
+                        format,
+                        self.bucket.clone(),
+                        self.entry.token_data_id.clone(),
+                        new_img,
+                    )
+                    .await
+                    {
+                        Ok(filename) => {
+                            self.model.cdn_animation_uri =
+                                Some(format!("http://34.160.26.161/{}", filename));
+                            info!(
+                                last_transaction_version = self.entry.last_transaction_version,
+                                "Successfully saved image"
+                            );
+                        },
+                        Err(e) => error!(
+                            last_transaction_version = self.entry.last_transaction_version,
+                            "{}",
+                            e.to_string()
+                        ),
+                    }
+                },
+                Err(e) => {
+                    // Do not increment retry count for animation
```

Comment:
Because for most NFTs we don't expect there to be an animation. On top of that, we directly save the raw animation (don't perform optimizations). Should I still have a counter for that?

Reply:
Dealing with this next PR